rfq/order.go

```diff
@@ -167,6 +184,29 @@ func (c *AssetSalePolicy) CheckHtlcCompliance(
 	return nil
 }
 
+// TrackAcceptedHtlc accounts for the newly accepted htlc. This may affect the
+// acceptance of future htlcs.
+func (c *AssetSalePolicy) TrackAcceptedHtlc(htlc lndclient.InterceptedHtlc) {
+	c.CurrentAmountMsat += htlc.AmountOutMsat
+
+	htlcIDStr := htlcIdentifierStr(
```

Comment:
We can use structs as map keys directly. And as long as there are no pointer values in a struct (meaning the struct can be directly compared with `==` without the pointers leading to side effects), things work out as expected. So we can just use the circuit key directly as the map key, no need to convert to string.